sdk/core/System.ClientModel/tests/client/ModelReaderWriter/Models/DiscriminatorSet/ModelX.cs

```diff
@@ -11,13 +11,14 @@ namespace System.ClientModel.Tests.Client.ModelReaderWriterTests.Models
 {
     public class ModelX : BaseModel, IJsonModel<ModelX>
     {
-        public ModelX()
-            : base(null)
+        public ModelX() : base(null)
         {
             Kind = "X";
+            Fields = new List<string>();
```

Comment:
Non-nullable properties must either be initialized or follow the assert pattern.